### PR TITLE
Anchor copy component

### LINKED ISSUE #149: Create and AnchorCopy component
I would like to have an AnchorCopy core component created with the ability to copy inserted page anchors. The component should encode the copied text, append it to the current url as a hash, and add it to the users clipboard.

### PR DESCRIPTION
**Pull request by jason-ransom-slalom**
This PR is for adding the `AnchorCopy` component. If approved, it will close #149 .

Screen Shot 2021-08-24 at 12 07 11 PM
Screen Shot 2021-08-24 at 12 07 22 PM

### CODE REVIEW

File: src/AnchorCopy/AnchorCopy.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    className,
+    disabled,
+    hashId,
+    onCopy,
+    title,
+    ...props
+}) => {
+    const hash = hashId || title;
+
+    const AnchorContentComponent = Boolean(AnchorContent) ? (
```
Comment (jason-ransom-slalom) [inline src/AnchorCopy/AnchorCopy.tsx:64]: This was added to allow the user the option to enhance the provided anchor (such as to add a tooltip, replace with an image, etc).

**Review by amirkibbar — APPROVED** [review]

File: src/lib/util/copy/__tests__/copy.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import copy from '../copy';
+
+describe('copy', () => {
+    it('copies the provided text to the clipboard', () => {
+        const writeText = jest.fn();
+
+        global.navigator.clipboard = {
```
Comment (adorack-tri) [inline src/lib/util/copy/__tests__/copy.test.js:7]: These tests that override `global.navigator.clipboard`... I think we should make sure to set this back to the original value after the test in order to ensure consistent behavior in any other tests that might use `writeText`.  Or am I missing something that would make this not a problem?

**Review by adorack-tri — CHANGES_REQUESTED** [review]

**Review by JonTheBurley — APPROVED** [review]

File: src/lib/util/copy/__tests__/copy.test.js (hunk 2 above)
Comment (jason-ransom-slalom) [reply src/lib/util/copy/__tests__/copy.test.js:13]: Thanks! I wasn't sure those changes carried between tests. Either way, I've added a reset after each that should suffice. Let me know if you see any further issues.

**Review by adorack-tri — APPROVED** [review]

File: src/lib/util/copy/__tests__/copy.test.js (hunk 2 above)
Comment (adorack-tri) [reply src/lib/util/copy/__tests__/copy.test.js:13]: I wasn't 100% sure if the changes carried either, but adding the reset does seem safer.  Thanks!

**Pull request merged**